tsl/src/compression/api.c

```diff
@@ -240,6 +240,23 @@ compress_chunk_impl(Oid hypertable_relid, Oid chunk_relid)
 					AccessShareLock);
 	LockRelationOid(catalog_get_table_id(ts_catalog_get(), CHUNK), RowExclusiveLock);
 
+	/*
+	 * Re-read the state of the chunk after all locks have been acquired and ensure
+	 * it is still uncompressed. Another process running in parallel might have
+	 * already performed compression while we were waiting for the locks to be
+	 * acquired.
+	 */
+	Chunk *chunk_state_after_lock = ts_chunk_get_by_relid(chunk_relid, true);
+	if (chunk_state_after_lock->fd.compressed_chunk_id != INVALID_CHUNK_ID)
```

Comment:
We cannot perform this operation if the chunk was frozen.
Probably should centralize all the eligibility logic in 
ts_chunk_validate_chunk_status_for_operation(...) and call that function here.
i.e. modify ts_chunk_validate_chunk_status_for_operation to account for the "compressed" state as well.

Reply:
Thanks, it is changed in the current version of the PR.